### PR TITLE
Move placement-new tests out of badinc.cc

### PR DESCRIPTION
**Pull request by kimgr**
This is separate enough that moving it into its dedicated test file was
reasonably trivial.

Complemented test cases to cover placement-new:
* of builtin type in normal code
* of user-defined type in normal code
* in a macro
* in an uninstantiated function template
* of an instantiated class template

In the process lost one seemingly pointless test that
'delete (((newed_int)));' does not have any effect.

### CODE REVIEW

**Comment by kimgr** [comment]
@adembo Please take a look, I thought I'd test your branch on top of this too.

File: tests/cxx/placement_new.cc [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  new (newed_int) int(4);
+
+  delete newed_int;
+}
+
+// Placement-new of user-defined type.
+void PlacementNewUserType() {
+  // IWYU: IndirectClass needs a declaration
+  // IWYU: IndirectClass is...*indirect.h
+  IndirectClass* icptr = new IndirectClass;
```
Comment (adembo) [inline tests/cxx/placement_new.cc:34]: Need to delete this at the end of the function.

Comment (kimgr) [reply tests/cxx/placement_new.cc:34]: Thanks, done. On another branch, coming soon.

**Comment by kimgr** [comment]
Abandoning this, covered in PR #784 784

**Pull request closed without merging**